test/chain-adapters/Optimism_Adapter.ts

```diff
@@ -0,0 +1,86 @@
+import { sampleL2Gas, amountToLp, mockTreeRoot, refundProposalLiveness, bondAmount } from "./../constants";
+import { ethers, expect, Contract, FakeContract, SignerWithAddress, createMock } from "../utils";
+import { getContractFactory, seedWallet } from "../utils";
+import { hubPoolFixture, enableTokensForLP } from "../HubPool.Fixture";
+import { constructSingleChainTree } from "../MerkleLib.utils";
+
+let hubPool: Contract, optimismAdapter: Contract, weth: Contract, dai: Contract, timer: Contract, mockSpoke: Contract;
+let l2Weth: string, l2Dai: string;
+let owner: SignerWithAddress, dataWorker: SignerWithAddress, liquidityProvider: SignerWithAddress;
+let l1CrossDomainMessenger: FakeContract, l1StandardBridge: FakeContract;
+
+const optimismChainId = 10;
+
+describe("Optimism Chain Adapter", function () {
+  beforeEach(async function () {
+    [owner, dataWorker, liquidityProvider] = await ethers.getSigners();
+    ({ weth, dai, l2Weth, l2Dai, hubPool, mockSpoke, timer } = await hubPoolFixture());
+    await seedWallet(dataWorker, [dai], weth, amountToLp);
+    await seedWallet(liquidityProvider, [dai], weth, amountToLp.mul(10));
+
+    await enableTokensForLP(owner, hubPool, weth, [weth, dai]);
+    await weth.connect(liquidityProvider).approve(hubPool.address, amountToLp);
```

Comment:
Do we not have a module in `utils` for all this setup for LP tokens and whitelisting routes in the hub pool?

Reply:
not for these but we could refactor accordingly. youy right that we do this approve/addliquidity call in a bunch of places.